### PR TITLE
Add endpoint to get most used categories for purchases

### PR DESCRIPTION
**Pull request by benymng**
* Add an endpoint to get the categories that are most used by a group

image

### CODE REVIEW

File: analytics/services.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from multiset.db_utils import execute_query
+from purchases.models import Purchase
+from purchases.services import get_purchases_by_group_id
+from collections import defaultdict
+
+
+def get_purchase_category_count(group_id: int):
+    rows = get_purchases_by_group_id(group_id)
```
Comment (weldon-s) [inline analytics/services.py]: Should this logic be contained in the query instead? Ordinarily I'd agree w putting it here but db course yk

**Review by weldon-s — COMMENTED** [review]
LGTM! Only thing I'm not sure of is how much logic we should be adding on the SQL side

File: analytics/services.py (hunk 1 above)
Comment (benymng) [reply analytics/services.py]: Yes good suggestion! I just updated this

**Review by emmahuangg — APPROVED** [review]
LGTM

**Pull request merged**